### DISCUSSION TITLE
All Diffusers having almost same BBoxLow and BboxHigh

### DISCUSSION THREAD

**Discussion by tusharsable** [iTwin/itwinjs-core#2748] [Q&A] [answered]
We want to show flowlines of air through the diffusers when the diffure are selected. we have flowline point array . we need to check isf startig point of flowline lies within the bbox of the diffuser and show them accordingly. But when I queried for Bbox I am getting almost same values for all diffusers. 
![image](https://user-images.githubusercontent.com/66006244/142575897-65d8e61e-6a28-46a6-b505-cf540ea979b8.png)
The origin for individual diffusers is different though
![image](https://user-images.githubusercontent.com/66006244/142575983-acbb18e9-242b-43b5-b4fd-29e8ccce12ef.png)

Is there any solution for this or what could be the reason

**Comment by pmconne** [comment, marked as answer]
BBoxLow and BBoxHigh describe the bounding box in the element's **local** coordinate space - i.e., its [ElementAlignedBox3d](https://www.itwinjs.org/reference/imodeljs-common/geometry/elementalignedbox3d/).
If you want its bounding box in **world** coordinates - i.e., its [AxisAlignedBox3d](https://www.itwinjs.org/reference/imodeljs-common/geometry/axisalignedbox3d/) you must apply the element's transform.
See for example [Placement3d.calculateRange](https://www.itwinjs.org/reference/imodeljs-common/geometry/placement3d/).
